src/components/Profile/ProfileBanner.tsx

```diff
@@ -13,14 +13,23 @@ const ProfileBanner = ({user}: {user: FetchedUserProfile | undefined}) => {
   const authContext = useContext(AuthContext);
   const [newPermission, setNewPermission] = useState<number>(0);
   const [actionDescription, setActionDescription] = useState<string>('');
+  const [passwordCheck, setPasswordCheck] = useState<string>('');
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleClick = () => setShowPassword(!showPassword);
 
   const handleEditPermission = () => {
     setEditPermissionModal(false);
-    console.log('will change permission to ' + newPermission);
+
+    if (user != undefined) {
+      console.log('will change permission to ' + UserStatus[newPermission]);
+      authContext.user?.userObject.editOtherStatus(passwordCheck, user?.userObject, newPermission, actionDescription);
+      user.status = newPermission;
+    }
   };
 
-  const managerSwap = () => {
-    console.log('Will make this user manager and remove manager status from current manager');
+  const getStatusTitle = (status: number) => {
```

Comment:
you should hoist this outside of the ProfileBanner function so that it doesn't render the component again when `status` changes.